File: languages/python/bloock/entity/identity_v2/bjj_issuer_key.py
import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.issuer_key_args import IssuerKeyArgs
from bloock.entity.identity_v2.issuer_key import IssuerKey


class BjjIssuerKey(IssuerKey):

    def __init__(self, args: IssuerKeyArgs) -> None:
        self.args = args

    def to_proto(self) -> proto.IssuerKey:
        local_key = None
        if self.args.local_key is not None:
            local_key = self.args.local_key.to_proto()

        managed_key = None
        if self.args.managed_key is not None:
            managed_key = self.args.managed_key.to_proto()

        return proto.IssuerKey(
            local_key=local_key,
            managed_key=managed_key,
        )

File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/BjjIssuerKey.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;

public class BjjIssuerKey implements IssuerKey {
  IssuerKeyArgs args;

  public BjjIssuerKey(IssuerKeyArgs args) {
    this.args = args;
  }

  @Override
  public IdentityEntitiesV2.IssuerKey toProto() {
    IdentityEntitiesV2.IssuerKey.Builder builder = IdentityEntitiesV2.IssuerKey.newBuilder();

    if (this.args.localKey != null) {
      builder.setLocalKey(this.args.localKey.toProto());
    }

    if (this.args.managedKey != null) {
      builder.setManagedKey(this.args.managedKey.toProto());
    }

    return builder.build();
  }
}
